src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.General.cs

```diff
@@ -82,6 +93,12 @@ private static partial class GeneralLog
         [LoggerMessage(41, LogLevel.Warning, "One or more of the following response headers have been removed because they are invalid for HTTP/2 and HTTP/3 responses: 'Connection', 'Transfer-Encoding', 'Keep-Alive', 'Upgrade' and 'Proxy-Connection'.", EventName = "InvalidResponseHeaderRemoved")]
         public static partial void InvalidResponseHeaderRemoved(ILogger logger);
 
-        // Highest shared ID is 63. New consecutive IDs start at 64
+        [LoggerMessage(64, LogLevel.Warning, "HTTP/2 is not enabled for {Endpoint}. The endpoint is configured to use HTTP/1.1 and HTTP/2, but TLS is not enabled. HTTP/2 requires TLS application protocol negotiation. Connections to this endpoint will use HTTP/1.1.", EventName = "Http2DisabledWithHttp1AndNoTls")]
+        public static partial void Http2DisabledWithHttp1AndNoTls(ILogger logger, EndPoint endPoint);
+
+        [LoggerMessage(65, LogLevel.Information, "HTTP/3 is not enabled for {Endpoint}. HTTP/3 requires TLS. Connections to this endpoint will use HTTP/1.1.", EventName = "Http3DisabledWithHttp1AndNoTls")]
+        public static partial void Http3DisabledWithHttp1AndNoTls(ILogger logger, EndPoint endPoint);
```

Comment:
Why are the log levels different? The conditions are the same. Use Warning for both?

Reply:
I went with Info for http/3 because it _never_ works without TLS, whereas I understand http/2 can work without TLS with some sort of pre-arranged setup (and I was tacitly assuming that I was the only one who didn't know that).

Reply:
That seems backwards. With that logic I'd expect this to be an Error 😁. Let's stick with Warning for both.

Reply:
Will do, but note that there _is_ an error just below my change (when 1.1 isn't available as a fallback).